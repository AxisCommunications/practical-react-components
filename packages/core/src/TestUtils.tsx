import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
