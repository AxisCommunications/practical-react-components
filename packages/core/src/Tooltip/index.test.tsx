import React from 'react'
import 'jest-styled-components'

import { Tooltip, ExpandedTooltipTypography } from '.'
import { TestRender, TestTextWithForwardRef } from '../TestUtils'

describe('Tooltip', () => {
  test('Default', () => {
    const tree1 = TestRender(
      <Tooltip text="test">
        <TestTextWithForwardRef />
      </Tooltip>
    )
    expect(tree1).toMatchSnapshot()
  })

  test('Expanded', () => {
    const tree1 = TestRender(
      <Tooltip
        variant="expanded"
        tipTitle="title"
        contents={
          <ExpandedTooltipTypography>
            Expanded tooltip test text
          </ExpandedTooltipTypography>
        }
      >
        <TestTextWithForwardRef />
      </Tooltip>
    )
    expect(tree1).toMatchSnapshot()

    const tree2 = TestRender(
      <Tooltip
        variant="expanded"
        tipTitle="title"
        extraInfo="2020-08-31"
        contents={
          <ExpandedTooltipTypography>
            Expanded tooltip test text
          </ExpandedTooltipTypography>
        }
      >
        <TestTextWithForwardRef />
      </Tooltip>
    )
    expect(tree2).toMatchSnapshot()

    const tree3 = TestRender(
      <Tooltip
        variant="expanded"
        placement="left-right"
        tipTitle="title"
        contents={
          <ExpandedTooltipTypography>
            Expanded tooltip test text
          </ExpandedTooltipTypography>
        }
      >
        <TestTextWithForwardRef />
      </Tooltip>
    )
    expect(tree3).toMatchSnapshot()
  })
})
