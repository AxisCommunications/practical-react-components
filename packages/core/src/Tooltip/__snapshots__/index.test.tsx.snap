// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Default 1`] = `
.c3 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c2 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <span>
        Test
      </span>
    </div>
  </div>
  <div>
    <div
      className="c2"
      id="layer-2"
    >
      <div
        className="c3"
      />
    </div>
  </div>
</div>
`;

exports[`Tooltip Expanded 1`] = `
.c3 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c2 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-3"
    >
      <span>
        Test
      </span>
    </div>
  </div>
  <div>
    <div
      className="c2"
      id="layer-4"
    >
      <div
        className="c3"
      />
    </div>
  </div>
</div>
`;

exports[`Tooltip Expanded 2`] = `
.c3 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c2 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-5"
    >
      <span>
        Test
      </span>
    </div>
  </div>
  <div>
    <div
      className="c2"
      id="layer-6"
    >
      <div
        className="c3"
      />
    </div>
  </div>
</div>
`;

exports[`Tooltip Expanded 3`] = `
.c3 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c2 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-7"
    >
      <span>
        Test
      </span>
    </div>
  </div>
  <div>
    <div
      className="c2"
      id="layer-8"
    >
      <div
        className="c3"
      />
    </div>
  </div>
</div>
`;
